xmpp/xmpp.go

```diff
@@ -88,16 +92,22 @@ func (c *Conn) Auth(user, pass, resource string) {
 	fmt.Fprintf(c.outgoing, xmlIqSet, id(), NsIqAuth, user, pass, resource)
 }
 
+func (c *Conn) Oauth(token, resource string) {
+	msg := "\x00" + token + "\x00" + resource
+	b64 := base64.StdEncoding.EncodeToString([]byte(msg))
+	fmt.Fprintf(c.outgoing, xmlOauth, b64)
+}
+
 func (c *Conn) Features() *features {
 	var f features
 	c.incoming.DecodeElement(&f, nil)
 	return &f
 }
 
 func (c *Conn) Next() (xml.StartElement, error) {
+	for {
```

Comment:
Moving the variable declaration helped me troubleshoot as the element was reinitiatlized on each iteration.  Otherwise, the first time an element is encountered, the variable is assigned and never changed. On a subsequent call to Token() that might lead to an error, element is still the element previously assigned and is returned, but instead 'element' should have been nil here.

At least that's what I think I was doing when I was troubleshooting oauth access.  If that doesn't make sense to you, feel free to ignore this line change. :-)